dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1236,6 +1236,17 @@ void InterpreterSelectQuery::executeMergeSorted(Pipeline & pipeline)
     /// If there are several streams, then we merge them into one
     if (pipeline.hasMoreThanOneStream())
     {
+        /// Unify streams in case they have different headers.
+        auto first_header = pipeline.streams.at(0)->getHeader();
```

Comment:
Maybe unify this code with same for Union stream?

Reply:
Yes, you can add unification in Union stream.